cmd/storagenode-updater/main.go

```diff
@@ -119,17 +117,21 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 	}()
 
 	loopFunc := func(ctx context.Context) (err error) {
-		if err := update(ctx, runCfg.BinaryLocation, runCfg.ServiceName, renameStoragenode); err != nil {
+		if err := update(ctx, runCfg.BinaryLocation, runCfg.ServiceName); err != nil {
 			// don't finish loop in case of error just wait for another execution
 			log.Println(err)
 		}
 
 		// TODO: enable self-autoupdate back when having reliable recovery mechanism
-		// updaterBinName := os.Args[0]
-		// if err := update(ctx, updaterBinName, updaterServiceName, renameUpdater); err != nil {
-		// 	// don't finish loop in case of error just wait for another execution
-		// 	log.Println(err)
-		// }
+		//updaterBinName, err := os.Executable()
+		//if err != nil {
+		//	log.Println(err)
+		//} else {
+		//	if err := update(ctx, updaterBinName, updaterServiceName); err != nil {
+		//		// don't finish loop in case of error just wait for another execution
+		//		log.Println(err)
+		//	}
+		//}
```

Comment:
Why is this change?

Reply:
I was unaware of `os.Executable` before and it seemed to be the more correct way to achieve this. Additionally, the signature of `update` changed so I updated the call accordingly so that the commented code will work when uncommented.

Reply:
what about this part:
```
if err != nil {
	log.Println(err)
} else {
	...
}
```

Reply:
`os.Executable` can return an error where accessing `os.Args[0]` can't. In that case I think we still don't want to stop the loop because we want storagenode to keep updating.

Reply:
oh, ok... But then what's the benefit of using `os.Executable` if `os.Args[0]` returns the same thing without the risk of error?

Reply:
I don't have a good answer. It's not completely clear to me right now the specific differences between the two and which is preferable here. My thinking was that it seemed reasonable to use a named function that explicitly identifies itself as the thing we want rather than the less explicit `os.Args[0]`. Both are used in the codebase; however, `os.Args[0]` is used more frequently.

It seems to be the case that the difference between the two options have a lot to do with how the process was started and in this case they may always be identical. If you want to change it back that's fine with me.

Reply:
I'd prefer to change it back to avoid unexpected issues.

Reply:
Done